templates/catalog/_partials/miniatures/product.tpl

```diff
@@ -168,6 +168,26 @@
               {/block}
             </div>
 
+            {if $product.add_to_cart_url}
+              <form action="{$urls.pages.cart}" method="post" class="d-flex align-items-center mt-3">
+                <input type="hidden" value="{$product.id_product}" name="id_product">
+                <input type="hidden" name="token" value="{$static_token}" />
```

Comment:
This won't work for the product lists, there's a need to have this done via JS to ensure the token is correct.

CC @Oksydan

Reply:
Yes, @kpodemski is right. 
Block might be cached and we should make sure that token provided to request is correct @NeOMakinG.
We could simply replace `static_token` value with `prestashop.static_token` global variable here https://github.com/PrestaShop/PrestaShop/blob/develop/themes/_core/js/cart.js#L101 with use of https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams or regex replace (might be tricky).

Reply:
@Oksydan do you think it's possible to somehow manage it on the theme side?

Reply:
@NeOMakinG you can make some dirty tricks like:

```js
document.querySelectorAll('[name="token"]').forEach(el => {
  el.value = prestashop.static_token
})
```
That's major disadvantage of using core.js 😕

Reply:
> @NeOMakinG you can make some dirty tricks like:
> 
> ```js
> document.querySelectorAll('[name="token"]').forEach(el => {
>   el.value = prestashop.static_token
> })
> ```
> 
> That's major disadvantage of using core.js 😕
> 
> EDIT: I just realized that you can ofc create you own function that will handle add to cart process but that's some extra code to maintain with core.js changes. Still no recommended.

I would probably prefer that dirty tricks over changing the corejs behavior on a minor version